pilot/pkg/security/authn/v1beta1/policy_applier.go

```diff
@@ -389,10 +389,14 @@ func convertToEnvoyJwtConfig(jwtRules []*v1beta1.JWTRule, push *model.PushContex
 }
 
 // buildLocalJwks builds local Jwks by fetching the Jwt Public Key from the URL passed if it is empty.
-func buildLocalJwks(jwksURI, jwtPubKey string) *envoy_jwt.JwtProvider_LocalJwks {
+func buildLocalJwks(jwksURI, jwtPubKey string, jwtKeyResolver *model.JwksResolver) *envoy_jwt.JwtProvider_LocalJwks {
 	if jwtPubKey == "" {
 		var err error
-		jwtPubKey, err = model.GetJwtKeyResolver().GetPublicKey(jwksURI)
+
+		if jwtKeyResolver == nil {
+			log.Errorf("JWT key resolver not found")
```

Comment:
When jwtKeyResolver == nil, this function will call the following two statements and the second statement will crash since jwtKeyResolver is nil.
log.Errorf("JWT key resolver not found")
jwtPubKey, err = jwtKeyResolver.GetPublicKey(jwksURI)

Reply:
Thanks! Added nil key resolver handling by reusing the fake jwks logic.

Reply:
When jwtKeyResolver == nil, why does the function use a fake public key (jwtPubKey = createFakeJwks(jwksURI)) instead of returning an error?

Reply:
I'm personally OK with an error. However, there are two concerns:
1. we probably need to propagate the error till the adding of jwt filters, as the current logic will always add one if it's not nil (and we can't simply make it nil) - this may change the current a bit;
2. supposing the behavior should align w/ GetPublicKey() fail - as they both are something that need to be informed at the control plane.

WDYT? @lei-tang

Reply:
When jwtKeyResolver == nil, the function returns a fake key. Why will the fake key work?

Reply:
OK, thanks! Updated by returning an error.

Reply:
> When jwtKeyResolver == nil, why does the function use a fake public key (jwtPubKey = createFakeJwks(jwksURI)) instead of returning an error?

Hi Lei, we are generating a fake jwt to reject the request. It's implemented in this [PR](https://github.com/istio/istio/pull/25934)  and discussed in this [issue comment.](https://github.com/istio/istio/issues/24629#issuecomment-645019117)